faiss/IndexHNSW.cpp

```diff
@@ -647,6 +647,155 @@ IndexHNSWFlat::IndexHNSWFlat(int d, int M, MetricType metric)
     is_trained = true;
 }
 
+IndexHNSWFlat::IndexHNSWFlat(int d, int M, IndexFlat* storage)
+        : IndexHNSW(storage, M) {
+    own_fields = true;
```

Comment:
In general when storage is passed as a parameter, the fields are not owned by default.